worker/games.py

```diff
@@ -544,6 +573,37 @@ def parse_options(s):
     os.remove(zipball)
     os.chmod(cutechess, os.stat(cutechess).st_mode | stat.S_IEXEC)
 
+  # clean up old networks (keeping the 20 most recent)
+  networks = glob.glob(os.path.join(testing_dir, 'nn-*.nnue'))
+  if len(networks) > 20:
+    networks.sort(key=os.path.getmtime)
+    for old_net in networks[:-20]:
+      try:
+         os.remove(old_net)
+      except:
+         print('Note: failed to remove an old network ' + str(old_net))
+         pass
+
+  # Add EvalFile with full path to cutechess options, and download networks if not already existing
+  net_base = required_net(base_engine)
+  if net_base:
+     base_options = base_options + ["option.EvalFile=%s"%(os.path.join(testing_dir, net_base))]
+  net_new = required_net(new_engine)
+  if net_new:
+     new_options = new_options + ["option.EvalFile=%s"%(os.path.join(testing_dir, net_new))]
+
+  for net in [net_base, net_new]:
+    if net:
+      try:
+        if not os.path.exists(os.path.join(testing_dir, net)) or not validate_net(testing_dir, net):
+           download_net(testing_dir, net)
+           if not validate_net(testing_dir, net):
+              raise Exception('Failed to validate the network: %s ' % (net))
+      except Exception as e:
+        sys.stderr.write('Exception from downloading network: %s \n' % (net))
+        print(e)
```

Comment:
You could rethrow the Exception here (`raise`), ore just remove the try/except code completely. EDIT: I see you just did :-)

I expect that the error will be handled correctly in the caller. Just like the `raise` in line 601

Reply:
Thanks! Yes, seems to work.